tests/simulator_test.cpp

```diff
@@ -37,22 +37,23 @@ class EnergyCollectorTest : public ::testing::Test {
   enum class HitResult { ENERGY_COLLECTORS_EMPTY, NO_HIT, HIT };
 
 protected:
-  std::vector<std::unique_ptr<objects::EnergyCollector>> energyCollectors;
   MockModel nonEmptyModel, emptyModel;
 
   // performs ray hit at energy Collectors by modifying hitData.
   // Returns:
   // ENERGY_COLLECTORS_EMPTY - when no energyCollectors where assigned
   // NO_HIT - when there was no hit
   // HIT - when hit occurred;
-  [[nodiscard]] HitResult performHitCollector(const Ray &ray, float frequency,
-                                              RayHitData *hitData) {
+  [[nodiscard]] HitResult performHitCollector(
+      const Ray &ray,
+      std::vector<std::unique_ptr<objects::EnergyCollector>> *energyCollectors,
+      RayHitData *hitData) {
 
-    if (energyCollectors.empty()) {
+    if (energyCollectors->empty()) {
```

Comment:
Why do you still need this? Will you ever assert on this in the test? Likely no, you are already checking if the number of collectors it equal to N, so this is useless, and is making the code much more complicated than it needs to be (with bool).

It good to write test code like:

```
collectors = buildCollectors(...);
ASSERT_EQ(12, collectors.size());

ASSERT_TRUE(checkHit(collectors, ray, &hitData));
```

Here you check the collectors size just after it was build, and test reads naturally.